opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -529,16 +529,13 @@ class LiveOilPvt
             const Evaluation& f = RsTable.eval(pSat, /*extrapolate=*/true) - Rs;
             const Evaluation& fPrime = RsTable.evalDerivative(pSat, /*extrapolate=*/true);
 
-            const Evaluation& delta = f/fPrime;
-
-            if (!Opm::isfinite(delta)) {
-                 std::stringstream errlog;
-                 errlog << "Obtain delta with non-finite value for Rs = " << Rs << " in the "
-                        << i << "th iteration during finding saturation pressure iteratively";
-                 OpmLog::problem("wetgas NaN delta value", errlog.str());
-                 OPM_THROW_NOLOG(NumericalProblem, errlog.str());
+            //If the derivative is zero Newton will not converge, so simply return our best guess instead
+            if (Toolbox::scalarValue(fPrime) == 0.0) {
+                return pSat;
```

Comment:
My initial tests with this were not very successful. I'm running a simulation now to see if it is actually better or not.